src/app/contribute/guidelines/components/index.html

```diff
@@ -0,0 +1,240 @@
+    </li>
+  </ol>
+
+  <stache-page-anchor>
+    Write code
+  </stache-page-anchor>
+
+  <ol>
+    <li>
+      Launch a command prompt, <stache-code>cd</stache-code> to the folder where you cloned your branch, and run <stache-code>npm run watch</stache-code>.
+    </li>
+    <li>
+      Launch a second command prompt, <stache-code>cd</stache-code> to the folder where you cloned your branch, and run <stache-code>skyux serve</stache-code>. This launches the SKY UX component demos page where you write the demo components to test your code.
+    </li>
+    <ul>
+      <li>
+        SKY UX demos are authored with <a href="https://github.com/blackbaud/skyux-sdk-cli">SKY UX SDK CLI</a>. If this is your first time using SKY UX Builder, you must install the CLI as a global NPM package before you run <stache-code>skyux serve</stache-code>:
+      </li>
+      <stache-code>npm i -g @skyux-sdk/cli</stache-code>
+    </ul>
+    <li>
+      When you save code changes, the code compiles, static code analysis is performed, unit tests run, and the page refreshes with the latest changes.
+    </li>
+    <p>
+      Keep in mind that when you create components, you must add public exports through <stache-code>./src/core.ts</stache-code> and <stache-code>./src/demo.ts</stache-code> to allow users to import your component modules. The public exports provide entry points for the bundles that SKY UX deploys.
+    </p>
+    <p>
+      The recommended change detection strategy for SKY UX components is to use <a href="https://angular.io/api/core/ChangeDetectionStrategy">the <stache-code>OnPush</stache-code> property instead of <stache-code>Default</stache-code></a>.
+    </p>
+  </ol>
+
+  <stache-page-anchor>
+    Write documentation
+  </stache-page-anchor>
+
+  <p>
+    We recommend that you create reference documentation for the <a stacheRouterLink="/components">Components section</a> of the SKY UX website. Component documentation should be extensive enough to explain the features of your components, including all properties and events. In the <a href="https://github.com/blackbaud/skyux2-docs">SKY UX docs repo</a>, you create subfolders for your components in the <stache-code>src/app/components</stache-code> folder and document them in index.html files. To create action buttons for the Components section landing page, you add entries in the <stache-code>demo-components.service.ts</stache-code> file.
+  </p> 
+  
+  <p>
+    If you do not document your components, the SKY UX team will create the documentation, but we request that you <a href="https://github.com/blackbaud/skyux2-docs/issues">file an issue</a>. However, we strongly encourage you to create your own documentation because it can be useful to uncover issues with your components.
```

Comment:
Did we agree that rough documentation (or bullet points) should be added to the original issue, instead of creating a docs-specific issue? Do you remember @blackbaud-johnly?

Reply:
I didn't address this one simply because I'm not sure what the original decision was.

Reply:
Sorry for the nonresponse on this. I missed the comment somehow. ... Anyway, I don't remember any decision about recommending bullet points to the original issue instead of a docs-specific issue. My gut feeling is that we should request the docs issue. We are going to need that issue anyway. But we can always revise these recommendations if you want. For now, I think we should just leave it as is.